### PR TITLE
Unify 3 agent queries into 1

### PR DESCRIPTION
**Pull request by maackle**
### INCLUDED CHANGES:
- The following API messages have been consolidated into a single `query_agents`:
    - `query_gossip_agents`
    - `handle_query_agent_info_signed`
    - `handle_query_agent_info_signed_near_basis`
- The holochain side is unchanged. The single query gets split into 3 paths by duck typing on the single query object to infer the intent.
    - Ideally at some point we could unify this on the Holochain side too, but it's not so important.
    - This is a kind of dubious thing I did, because we have to be a bit careful about changes on the Kitsune side which could caues the Holochain side to panic (though these panics will not be rare, we'll know about them immediately if they happen)
- I had to change the return type of `query_gossip_agents` to return a full `AgentInfoSigned` instead of the subset of data it was returning, to be able to unify the results with the other two queries.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info

### CODE REVIEW

File: crates/holochain_p2p/src/spawn/actor.rs [hunk 1]
```diff
@@ -635,67 +626,57 @@ impl kitsune_p2p::event::KitsuneP2pEventHandler for HolochainP2pActor {
...
+                    .query_agent_info_signed(h_space, agents, space)
+                    .await?),
+
+                // If none of the above match, we have no implementation for such a query
+                // and must fail
+                tuple => unimplemented!(
+                    "Holochain cannot interpret the QueryAgentsEvt data as given: {:?}",
+                    tuple
+                ),
+            }
```
Comment (maackle) [inline crates/holochain_p2p/src/spawn/actor.rs:679]: This is the questionable part

File: crates/holochain_p2p/src/spawn/actor.rs [hunk 2]
```diff
@@ -635,67 +627,58 @@ impl kitsune_p2p::event::KitsuneP2pEventHandler for HolochainP2pActor {
...
+                (agents, None, None, None, None) => Ok(evt_sender
+                    .query_agent_info_signed(h_space, agents, space)
+                    .await?),
+
+                // If none of the above match, we have no implementation for such a query
+                // and must fail
+                tuple => unimplemented!(
+                    "Holochain cannot interpret the QueryAgentsEvt data as given: {:?}",
+                    tuple
+                ),
```
Comment (neonphog) [inline crates/holochain_p2p/src/spawn/actor.rs:680]: hehe. yep, this works as a first iteration : )

File: crates/holochain/src/conductor/p2p_agent_store.rs [hunk 3]
```diff
@@ -83,7 +83,7 @@ pub fn list_all_agent_info_signed_near_basis(
     environ: EnvWrite,
     _kitsune_space: Arc<kitsune_p2p::KitsuneSpace>,
     basis_loc: u32,
-    limit: u32,
+    limit: usize,
```
Comment (neonphog) [inline crates/holochain/src/conductor/p2p_agent_store.rs:86]: Why switch to a type that is a different size on different systems? I personally feel like the bit-size consistency is worth the occasional `as usize` when needed.

**Review by neonphog — APPROVED** [review]
:+1:

File: crates/holochain/src/conductor/p2p_agent_store.rs (hunk 3 above)
Comment (maackle) [reply crates/holochain/src/conductor/p2p_agent_store.rs:86]: Fair enough, I'll change it back

**Review by freesig — APPROVED** [review]
Looks good

**Pull request merged**